trees_emit.h

```diff
@@ -53,9 +53,15 @@ extern Z_INTERNAL const int base_dist[D_CODES];
         bi_buf = val;\
         bi_valid = len;\
     } else {\
-        bi_buf |= val << bi_valid;\
+        if (bi_valid < BIT_BUF_SIZE) {\
+            bi_buf |= val << bi_valid;\
+        }\
         put_uint64(s, bi_buf);\
-        bi_buf = val >> (BIT_BUF_SIZE - bi_valid);\
+        if (bi_valid < BIT_BUF_SIZE) {\
+            bi_buf = val >> (BIT_BUF_SIZE - bi_valid);\
+        } else {\
+            bi_buf = val;\
```

Comment:
Here `bi_valid = len;` will cause static analyzer to trip on line 49 during next call... I suspect `t_len` might be either negative or very large positive number that results `total_bits` to wrap around to less than `BIT_BUF_SIZE` while `bi_valid` is 64 (or more).

We assume that `total_bits` is between 64 and 128 if we reach here.